src/storage/secondary/block/primitive_nullable_block_builder.rs

```diff
@@ -52,6 +53,14 @@ impl<T: PrimitiveFixedWidthEncode> BlockBuilder<T::ArrayType>
         !self.data.is_empty() && self.estimated_size() + 1 + T::WIDTH > self.target_size
     }
 
+    fn distinct_count(&self) -> usize {
+        let mut distinct_values = HashSet::<&[u8]>::new();
+        for item in self.data.chunks(T::WIDTH) {
+            distinct_values.insert(item);
```

Comment:
This seems to ignore `None` value? Maybe we should also take the bitmap into account.